source/administration/plugins.rst

```diff
@@ -12,12 +12,68 @@ Consider using a plugin in the following scenarios:
+
+.. image:: https://user-images.githubusercontent.com/915956/66891467-1b18eb80-ef9e-11e9-9de3-37a3c5899bd8.png
+
+Installing a Plugin
+~~~~~~~~~~~~~~~~
+
+When a new plugin becomes available on the marketplace, a new listing with a button labeled **Install** will be displayed.  After clicking **Install**, the latest plugin binary is downloaded from it's respective github repository and installed on the server.  If there is a cluster present, the plugin will be automatically distributed to each server automatically.   
+
+Configuring and enabling a plugin
+~~~~~~~~~~~~~~~
+
+Once a plugin has been installed (or pre-installed if it shipped with Mattermost binary release) there will be a "Configure" button shown next to the plugin listing.  
+
+1. Click on the button, you will be navigated to the plugins "Settings" page.  
+2. Fill in Plugin settings as required
+3. At the top of the page, set **Enable Plugin** setting to ``True``.  If this flag is not switched on, the plugin will never become active.
+4. Test out the plugin as needed
+
+
+Upgrading Plugins
+~~~~~~~~~~~~~~~~~
+
+In v5.16, the Marketplace only supports the installation of new plugins.  To upgrade a plugin, you will need to manually update it by downloading the binary file from the GitHub repository and uploading it in the **System Console > Plugin management** section.  In v5.18 (Dec 2019), the Marketplace will allow you to upgrade a plugin on demand when a new version becomes available.
+
+
+Marketplace Server
+-----
+
+There are two Marketplace settings in the **System Console > Plugin management** section:
+
+.. image:: https://user-images.githubusercontent.com/915956/66892854-94660d80-efa1-11e9-805c-a85223d43a07.png
+
+- **Enable Marketplace** Turns the Marketplace UI on or off for all users (including sysadmins)
+- **Marketplace URL** The location of the Marketplace Server you want to query for new plugins. Mattermost hosts a Plugin Marketplace for the community and is the default value for this field.  You can setup your own Marketplace server, see below.
+
+When you are first presented with the Marketplace, your Mattermost server will attempt to contect the Marketplace Server,run by Mattermost, and will be returned a list of available plugins that are appropriate based on the server version that is currently running.  Only your server version and search query is passed over to Mattermost Marketplace, we retain an anonymized record for product analytics whenever a new plugin is installed, unless you have opted out of `Telemetry <https://docs.mattermost.com/administration/telemetry.html>`__ previously. The `Plugin marketplace server code <https://github.com/mattermost/mattermost-marketplace>`__ is available as an open-source project and can be used to setup your own private Marketplace if desired.  
```

Comment:
" Only your server version and search query is passed over to Mattermost Marketplace"

I thought we would do local only search?